os/net/blemgr/bledev.c

```diff
@@ -33,9 +34,59 @@
 		}										\
 	} while (0)
 
+#define SCAN_INFO_BUFFER_SIZE 150
+static trble_scanned_device scan_info_buf[SCAN_INFO_BUFFER_SIZE] = { 0, };
+static int write_index = 0;
+static int read_index = 0;
+static sem_t g_scan_countsem;
+static volatile int g_run_process = 0;
+
+int trble_post_event(lwnl_cb_ble evt, void *buffer, int32_t buf_len)
+{
+	BLE_LOGV(TRBLE_TAG, "trble post event : %d\n", evt);
+	return lwnl_postmsg(LWNL_DEV_BLE, (int)evt, buffer, buf_len);
+}
+
+int trble_scan_data_enque(trble_scanned_device *info)
+{
+	if (g_run_process != 1) {
+		return -1;
+	}
+	int write_index_next = (write_index + 1) % SCAN_INFO_BUFFER_SIZE;
+	if (write_index_next == read_index) {
+		/* Queue is Full */
+		return -2;
+	}
+	memcpy(&scan_info_buf[write_index], info, sizeof(trble_scanned_device));
+	write_index = write_index_next;
+
+	sem_post(&g_scan_countsem);
+
+	return 0;
+}
+
+static int _bledev_handler(int argc, char *argv[])
+{
+	g_run_process = 1;
+	trble_scanned_device *scanned_device;
+
+	while (g_run_process) {
+		sem_wait(&g_scan_countsem);
+		int current_write_index = write_index;
```

Comment:
Is the variable `current_write_index` needed?
I think we can compare with `write_index` in the below.

Reply:
As you recommended, I fixed it